code/modules/power/lighting.dm

```diff
@@ -703,14 +703,13 @@
 		var/mob/living/carbon/human/H = user
 
 		if(istype(H))
-			var/datum/species/ethereal/eth_species = H.dna?.species
-			if(istype(eth_species))
-				var/datum/species/ethereal/E = H.dna.species
-				var/obj/item/organ/stomach/ethereal/stomach = H.getorganslot(ORGAN_SLOT_STOMACH)
-				if((E.drain_time > world.time) || !stomach)
+			var/obj/item/organ/stomach/maybe_stomach = H.getorganslot(ORGAN_SLOT_STOMACH)
+			if(istype(maybe_stomach, /obj/item/organ/stomach/ethereal))
```

Comment:
There are still a bunch of shitty istype checks like this, but I dont think they're significant enough to be worth reworking